### PR TITLE
Remove non-standard increase_allowance and decrease_allowance from ERC20

### LINKED ISSUE #728: Remove non-standard increase_allowance and decrease_allowance from ERC20
**🧐 Motivation**

The Starknet equivalent to the [recent ERC20 update of Solidity contracts](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4585). Having these methods is even less necessary on Starknet thanks to AA and multicall. 

Additional benefit of removing them is slightly reduced contract size.

Since the Cairo 2 contracts haven't been officially released, I think it's a good opportunity to squeeze this in 🤞

### PR DESCRIPTION
**Pull request by TAdev0**
Fixes #728 

This PR removes `increase_allowance` and `decrease_allowance` functions in ERC20 contracts (token and mocks)

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [x] Tried the feature on a public network

### CODE REVIEW

**Comment by TAdev0** [comment]
@martriay  oh yes I still need to modify the doc , I forgot to do it. Same for the other PR.

**Comment by TAdev0** [comment]
@martriay  now it should be good :)

**Comment by martriay** [comment]
Thanks! Please complete all items in the PR checklist.

**Comment by TAdev0** [comment]
@martriay  am i suppose to ''try the feature'' on a public network? If yes what should i do?

**Comment by martriay** [comment]
Since this is a feature removal PR, we need to deploy the ERC20 preset and verify:

- it works properly for regular operations
- removed features don't work

Note that this PR modifies a preset, so we need to update the class hashes as [indicated in the CONTRIBUTING guide](https://github.com/OpenZeppelin/cairo-contracts/blob/main/CONTRIBUTING.md#class-hashes).

**Comment by TAdev0** [comment]
Thank very much for all explanations 🙏

**Comment by TAdev0** [comment]
@martriay 

Class hash: 0x7c9b5a246fe83b0cd81de65daddb94b3803f94950db99bf2431bbfecf284642

ERC20 preset deployed on Goerli at: 0x1ceea6ba0ad2015c13cdca03185f615127523d8ece2293975354163479db3e9

If you try to run, for example : 
```
sncast --url https://free-rpc.nethermind.io/goerli-juno --account [your_account] --contract-address 0x1ceea6ba0ad2015c13cdca03185f615127523d8ece2293975354163479db3e9 --function "increase_allowance" --calldata 0x0 0x1 0x0
```

You will get the following error: 
```
command: invoke
error: Transaction execution error = TransactionExecutionErrorData { transaction_index: 0, execution_error: "reverted: Error in the called contract (0x039ccf2f422488ac967891266df98cbf293e6dcafe6cf42ba603d2821fd9e247):\nError at pc=0:81:\nGot an exception while executing a hint: Custom Hint Error: Entry point EntryPointSelector(StarkFelt(\"0x01d13ab0a76d7407b1d5faccd4b3d8a9efe42f3d3c21766431d4fafb30f45bd4\")) not found in contract.\nCairo traceback (most recent call last):\nUnknown location (pc=0:731)\nUnknown location (pc=0:677)\nUnknown location (pc=0:291)\nUnknown location (pc=0:314)\n" }
``` 

This means `increase_allowance' entrypoint doesn't exist anymore.

Same applies for 

```
sncast --url https://free-rpc.nethermind.io/goerli-juno --account [your_account] --contract-address 0x1ceea6ba0ad2015c13cdca03185f615127523d8ece2293975354163479db3e9 --function "decrease_allowance" --calldata 0x0 0x0 0x0
```

```
error: Transaction execution error = TransactionExecutionErrorData { transaction_index: 0, execution_error: "reverted: Error in the called contract (0x039ccf2f422488ac967891266df98cbf293e6dcafe6cf42ba603d2821fd9e247):\nError at pc=0:81:\nGot an exception while executing a hint: Custom Hint Error: Entry point EntryPointSelector(StarkFelt(\"0x03b076186c19fe96221e4dfacd40c519f612eae02e0555e4e115a2a6cf2f1c1f\")) not found in contract.\nCairo traceback (most recent call last):\nUnknown location (pc=0:731)\nUnknown location (pc=0:677)\nUnknown location (pc=0:291)\nUnknown location (pc=0:314)\n" }
```

**Review by andrew-fleming — COMMENTED** [review]
Hey, thanks for opening this PR! I left a few comments and suggestions. Would you mind also removing the safe allowance impls from index.adoc [here](https://github.com/TAdev0/OZ-cairo-contracts/blob/erc20_remove_increase_decrease_allowance/docs/modules/ROOT/pages/index.adoc?plain=1#L83-L89)?

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,3 +35,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Support higher tx versions in Account (#858)
 - Bump scarb to v2.4.1 (#858)
 - Add security section to Upgrades docs (#861)
+  
+### Removed
+
+- Non standard increase_allowance and decrease_allowance functions in ERC20 contract (#881)
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply CHANGELOG.md:41]: yes, it was a mistake, sorry

File: docs/modules/ROOT/pages/erc20.adoc [hunk 2]
```diff
@@ -18,7 +18,7 @@ See the {custom-decimals} guide.
 :erc20-supply: xref:/guides/erc20-supply.adoc[Creating ERC20 Supply]
 
 The following interface represents the full ABI of the Contracts for Cairo `ERC20` component.
-The interface includes the `IERC20` standard interface as well as non-standard functions like `increase_allowance`.
+The interface includes the `IERC20` standard interface.
```
Comment (martriay) [inline docs/modules/ROOT/pages/erc20.adoc:21]:
```suggestion
The interface includes the `IERC20` standard interface as well as `IERC20Camel` for backwards compatibility.
```

File: docs/modules/ROOT/pages/interfaces.adoc [hunk 3]
```diff
@@ -45,7 +45,7 @@ trait IERC20<TState> {
 
 === ABI traits
 
-They describe a contract's complete interface. This is useful to interface with a preset contract offered by this library, such as the ERC20 preset that includes non-standard functions like `increase_allowance`.
+They describe a contract's complete interface.
```
Comment (martriay) [inline docs/modules/ROOT/pages/interfaces.adoc:48]:
```suggestion
They describe a contract's complete interface. This is useful to interface with a preset contract offered by this library, such as the ERC20 preset that includes functions from different traits such as `IERC20` and `IERC20Camel`.
```

**Review by martriay — COMMENTED** [review]
Looking good to me!

**Review by andrew-fleming — APPROVED** [review]
LGTM!

**Review by ericnordelo — APPROVED** [review]
LGTM!

**Comment by martriay** [comment]
@TAdev0 please fix the merge conflicts and this is good to go.

**Comment by TAdev0** [comment]
@martriay  now it should be good!

**Review by martriay — APPROVED** [review]

**Comment by martriay** [comment]
Thanks!

**Pull request merged**